### PR TITLE
Increase timeout for windows jobs

### PR DESCRIPTION
**Pull request by jieru-hu**
## Motivation

We've been seeing some transient failures with preparing for the windows environment (timing out). With no better way to reproduce the failure, I'm increasing the timeout temporarily per circleCI support's suggestions.
some of the failures:
https://app.circleci.com/pipelines/github/facebookresearch/hydra/4729/workflows/598280d6-d2b9-4027-96f6-9e840c2c09bf/jobs/34255
https://app.circleci.com/pipelines/github/facebookresearch/hydra/4710/workflows/b4b392bf-8ee5-4b0b-b359-7e05ee5023a2/jobs/34139
https://app.circleci.com/pipelines/github/facebookresearch/hydra/4695/workflows/a9b93805-67e8-44fc-ae25-8e17fe3d6d64/jobs/34018
https://app.circleci.com/pipelines/github/facebookresearch/hydra/4750/workflows/5846959e-1515-4480-9e1a-5d077e24e3b0/jobs/34400

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebookresearch/hydra/blob/master/CONTRIBUTING.md)?

Yes

## Test Plan

CI
## Related Issues and PRs

(Is this PR part of a group of changes? Link the other relevant PRs and Issues here. Use https://help.github.com/en/articles/closing-issues-using-keywords for help on GitHub syntax)

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -146,7 +147,7 @@ jobs:
           py_version: << parameters.py_version >>
       - run:
           name: Testing Hydra
-          no_output_timeout: 10m
```
Comment (omry) [inline .circleci/config.yml:149]: This timeout is on the tests. Didn't we see the timeout while preparing the environment?

Comment (jieru-hu) [reply .circleci/config.yml:149]: 🤦🏻‍♀️  will fix

**Comment by jieru-hu** [comment]
Ok increasing timeout doesn't change anything.
Job failed with same failure
https://app.circleci.com/pipelines/github/facebookresearch/hydra/4784/workflows/9a8a1675-d666-4941-897d-5c26c5975d3a/jobs/34658

```
Download of JRE8x86.exe (69.53 MB) completed.
Hashes match.
C:\Users\circleci\AppData\Local\Temp\chocolatey\jre8\8.0.2710.9\JRE8x86.exe
Installing JRE 8.0.2710.9 32-bit
Installing JRE8...
JRE8 has been installed.
Downloading 64-bit installer
Downloading jre8 64 bit
  from 'https://javadl.oracle.com/webapps/download/AutoDL?BundleId=243737_61ae65e088624f5aaa0b1d2d801acb16'

Too long with no output (exceeded 30m0s): context deadline exceeded
```

**Comment by jieru-hu** [comment]
I think this might be the issue? https://github.com/chocolatey/choco/issues/1473

File: .circleci/config.yml [hunk 2]
```diff
@@ -140,19 +140,21 @@ jobs:
...
           name: Testing Hydra
           no_output_timeout: 10m
           command: |
-            $env:NOX_PYTHON_VERSIONS=<< parameters.py_version >>
-            $env:PYTHONIOENCODING="utf_8"
+            set NOX_PYTHON_VERSIONS="<< parameters.py_version >>"
+            set PYTHONIOENCODING="utf-8"
             conda activate hydra
             nox -s lint test_tools test_core test_jupyter_notebooks -ts
-            exit $LASTEXITCODE
```
Comment (omry) [inline .circleci/config.yml:155]: Can you propagate the nox exit code?

**Comment by omry** [comment]
> I think this might be the issue - [chocolatey/choco#1473](https://github.com/chocolatey/choco/issues/1473)

The workaround does not make sense:
https://github.com/chocolatey/choco-wiki/pull/191/files

They think it's interaction with Windows quickedit (you need to click the shell window for it to happen).
Switching to cmd (or better yet, bash - can still help).

File: .circleci/config.yml [hunk 3]
```diff
@@ -140,16 +141,19 @@ jobs:
...
           py_version: << parameters.py_version >>
       - run:
           name: Testing Hydra
           no_output_timeout: 10m
           command: |
-            $env:NOX_PYTHON_VERSIONS=<< parameters.py_version >>
-            $env:PYTHONIOENCODING="utf_8"
+            export NOX_PYTHON_VERSIONS="<< parameters.py_version >>"
+            export PYTHONIOENCODING="utf-8"
+            eval "$(conda shell.bash hook)"
```
Comment (omry) [inline .circleci/config.yml:156]: You have ` C:\tools\miniconda3\Scripts\conda.exe init powershell` here, which is why it does not work out of the box.
should probably change that to init for bash.

Comment (jieru-hu) [reply .circleci/config.yml:156]: Thx! I removed that.
`eval "$(conda shell.bash hook)"` this line seems to fix the init issue.

Comment (omry) [reply .circleci/config.yml:155]: you only need one of the two.
the init bash changes the shell to support conda permanently.
the eval you introduced changes the current session.

Comment (omry) [reply .circleci/config.yml:155]: I think `conda init bash` where we had `conda init powershell` is better.

**Comment by jieru-hu** [comment]
9aa970c - got a successful run https://app.circleci.com/pipelines/github/facebookresearch/hydra/4864/workflows/e5d1039e-62e8-4ef1-8911-caca6bcbbe21/jobs/35225

File: .circleci/config.yml (hunk 3 above)
Comment (jieru-hu) [reply .circleci/config.yml]: agreed. That was what i tried first. however the command failed with
`bash: line 1: C:toolsminiconda3Scriptsconda.exe: command not found`

failed run here https://app.circleci.com/pipelines/github/facebookresearch/hydra/4881/workflows/25dac692-1356-440d-af5c-ded57a2c62a9/jobs/35345
commit here https://github.com/facebookresearch/hydra/pull/1154/commits/f62120dcb36b9a21668c7bc9a1f10ea7469ab54d

found the workaround in this issue https://github.com/conda/conda-build/issues/3371

Comment (omry) [reply .circleci/config.yml:155]: `C:\tools\miniconda3\Scripts\conda.exe init powershell` has backslashes that are invalid in bash.
Try to switch to forward slashes.

Comment (jieru-hu) [reply .circleci/config.yml:155]: :O thx for the tip! will try it out.

**Comment by jieru-hu** [comment]
context from circleCI support team

------
After some investigations, we figured out there is a bad host in the repository of the JRE, and if the builds were routed to that host, the download speed become extremely slow, and the builds are failed.

Here is an example project. As far as we tested, the failed build used the same IP address to download JRE. If the builds use this host, they failed about 2 in 5 times.

Currently, the bad IP is 184.24.36.87.

https://github.com/ganezasan/circleci-windows-install-javaruntime-by-choco/blob/main/.circleci/config.yml

So, the way to avoid using this host, you can update hosts to specify the other IP address. Here is an example to update the hosts, could you try to add it before the download step?

      - run:
          shell: bash.exe
          name: add hosts
          command: |
            echo '23.35.74.212 sdlc-esd.oracle.com' >> C:/Windows/System32/drivers/etc/hosts
            tracert sdlc-esd.oracle.com

**Comment by jieru-hu** [comment]
It seems like this issue no longer exists (mostly like the bad host was fixed.)

closing this for now.

**Comment by omry** [comment]
Did we undo the ip address hack?

**Pull request closed without merging**